src/util/Helpers.hpp

```diff
@@ -37,4 +39,28 @@ QColor getRandomColor(const QString &userId);
 QString formatUserMention(const QString &userName, bool isFirstWord,
                           bool mentionUsersWithComma);
 
+template <typename T>
+std::vector<T> splitListIntoBatches(const T &list, int batchSize = 100)
+{
+    std::vector<T> batches;
+    int batchCount = std::ceil(list.size() / batchSize);
+    batches.reserve(batchCount);
+
+    auto it = list.cbegin();
+
+    for (int j = 0; j < batchCount; j++)
+    {
+        T batch;
+
+        for (int i = 0; i < batchSize && it != list.end(); i++)
+        {
+            batch.append(*it);
+            it++;
+        }
+        batches.emplace_back(std::move(batch));
```

Comment:
perhaps we can break early if batch is empty (in case of concurrent modifications making the list smaller); else emplace_back